File: C++/TicTacToe.cpp
#include <iostream>
using namespace std;

int main();
void game_board(char game_move[3][3]);
void validation(int row, int column, int miniumum_number, int maximum_number, int player, char game_move[3][3]);

void game_board(char game_move[3][3])
{
    system("clear");
	cout << "          |          |" << endl;
	cout << "     " << game_move[0][0] << "    |     " << game_move[0][1] << "    |     " << game_move[0][2] << "     " << endl;
	cout << "          |          |" << endl;
	cout << "__________|__________|__________" << endl;
	cout << "          |          |" << endl;
	cout << "          |          |" << endl;
	cout << "     " << game_move[1][0] << "    |     " << game_move[1][1] << "    |     " << game_move[1][2] << "     " << endl;
	cout << "          |          |" << endl;
	cout << "__________|__________|__________" << endl;
	cout << "          |          |" << endl;
	cout << "          |          |" << endl;
	cout << "     " << game_move[2][0] << "    |     " << game_move[2][1] << "    |     " << game_move[2][2] << "     " << endl;
	cout << "          |          |" << endl;
}

char player1(char game_move[3][3])
{
    int row, column;
    cout << "Enter a row number Player 1 (1-3): ";
    cin >> row;
    row--;
    cout << "Enter a column number Player 1 (1-3): ";
    cin >> column;
    column--;
    validation(row, column, 0, 3, 1, game_move);
    
    if ((game_move[0][0] == 'X' and game_move[0][1] == 'X' and game_move[0][2] == 'X') or 
        (game_move[1][0] == 'X' and game_move[1][1] == 'X' and game_move[1][2] == 'X') or 
        (game_move[2][0] == 'X' and game_move[2][1] == 'X' and game_move[2][2] == 'X') or 
        (game_move[0][0] == 'X' and game_move[1][0] == 'X' and game_move[2][0] == 'X') or 
        (game_move[0][1] == 'X' and game_move[1][1] == 'X' and game_move[2][1] == 'X') or 
        (game_move[0][2] == 'X' and game_move[1][2] == 'X' and game_move[2][2] == 'X') or 
        (game_move[0][0] == 'X' and game_move[1][1] == 'X' and game_move[2][2] == 'X') or 
        (game_move[0][2] == 'X' and game_move[1][1] == 'X' and game_move[2][0] == 'X'))
    {
        game_board(game_move);
        cout << "Player 1 won!" << endl;
        exit(0);
    }

    return game_move[row][column];
}

char player2(char game_move[3][3])
{
    int row, column;
    cout << "Enter a row number Player 2 (1-3): ";
    cin >> row;
    row--;
    cout << "Enter a column number Player 2 (1-3): ";
    cin >> column;
    column--;
    validation(row, column, 0, 3, 2, game_move);

    if ((game_move[0][0] == 'O' and game_move[0][1] == 'O' and game_move[0][2] == 'O') or 
        (game_move[1][0] == 'O' and game_move[1][1] == 'O' and game_move[1][2] == 'O') or 
        (game_move[2][0] == 'O' and game_move[2][1] == 'O' and game_move[2][2] == 'O') or 
        (game_move[0][0] == 'O' and game_move[1][0] == 'O' and game_move[2][0] == 'O') or 
        (game_move[0][1] == 'O' and game_move[1][1] == 'O' and game_move[2][1] == 'O') or 
        (game_move[0][2] == 'O' and game_move[1][2] == 'O' and game_move[2][2] == 'O') or 
        (game_move[0][0] == 'O' and game_move[1][1] == 'O' and game_move[2][2] == 'O') or 
        (game_move[0][2] == 'O' and game_move[1][1] == 'O' and game_move[2][0] == 'O'))
    {
        game_board(game_move);
        cout << "Player 2 won!" << endl;
        exit(0);
    }

    return game_move[row][column];
}

void validation(int row, int column, int miniumum_number, int maximum_number, int player, char game_move[3][3])
{
    if (cin.fail() or (row < miniumum_number or row > maximum_number) or (column < miniumum_number or column > maximum_number) or (game_move[row][column] == 'X' or game_move[row][column] == 'O'))
    {
        cin.clear();
        cin.ignore();
        game_board(game_move);
        if (player == 1)
        {
            player1(game_move);
        }
        else
        {
            player2(game_move);
        }
    }
    else
    {
        if (player == 1)
        {
            game_move[row][column] = 'X';
        }
        else
        {
            game_move[row][column] = 'O';
        }
    }
}

int main()
{
    char game_move[3][3] = { {'*', '*', '*'}, {'*', '*', '*'}, {'*', '*', '*'} };
    
    while (game_move[0][0] == '*' or game_move[0][1] == '*' or game_move[0][2] == '*' or game_move[1][0] == '*' or game_move[1][1] == '*' or game_move[1][2] == '*' or game_move[2][0] == '*' or game_move[2][1] == '*' or game_move[2][2] == '*')
    {
        game_board(game_move);
        player1(game_move);

        if (game_move[0][0] == '*' or game_move[0][1] == '*' or game_move[0][2] == '*' or game_move[1][0] == '*' or game_move[1][1] == '*' or game_move[1][2] == '*' or game_move[2][0] == '*' or game_move[2][1] == '*' or game_move[2][2] == '*')
        {
            game_board(game_move);
            player2(game_move);
        }
        cout << "Nobody wins!" << endl;
    }

    return 0;
}

File: Python3/TicTacToe.py
import os
game_move =[1, 2, 3, 4, 5, 6, 7, 8, 9]

# Prints Tic Tac Toe board on screen.
def game_board():
	os.system('cls' if os.name == 'nt' else 'clear')
	print("          |          |")
	print("     " + str(game_move[0]) + "    |     " + str(game_move[1]) + "    |     " + str(game_move[2]) + "     ")
	print("          |          |")
	print("__________|__________|__________")
	print("          |          |")
	print("          |          |")
	print("     " + str(game_move[3]) + "    |     " + str(game_move[4]) + "    |     " + str(game_move[5]) + "     ")
	print("          |          |")
	print("__________|__________|__________")
	print("          |          |")
	print("          |          |")
	print("     " + str(game_move[6]) + "    |     " + str(game_move[7]) + "    |     " + str(game_move[8]) + "     ")
	print("          |          |")

def player1():
	game_board()

# Determines if input from user is an integer and is 1-9. Also if slot on board is taken by an "X" or an "O".
	try:
		global action0
		action0 = int(input("\nWhat is your move player 1? (1-9) "))
	except ValueError:
		action0 = None
		pass
	if action0 is not None:
		if int(action0) >= 0 and int(action0) <= 10:
			if type(game_move[action0 - 1]) == int:
				game_move[int(action0) - 1] = "X"
			else:
				game_board()
				player1()
		else:
			game_board()
			player1()
	else:
		game_board()
		player1()

# Win condition for player 1.
	if game_move[0] == "X" and game_move[1] == "X" and game_move[2] == "X" or game_move[3] == "X" and game_move[4] == "X" and game_move[5] == "X" or game_move[6] == "X" and game_move[7] == "X" and game_move[8] == "X" or game_move[0] == "X" and game_move[3] == "X" and game_move[6] == "X" or game_move[1] == "X" and game_move[4] == "X" and game_move[7] == "X" or game_move[2] == "X" and game_move[5] == "X" and game_move[8] == "X" or game_move[0] == "X" and game_move[4] == "X" and game_move[8] == "X" or game_move[2] == "X" and game_move[4] == "X" and game_move[6] == "X":
		game_board()
		print("Player 1 wins!")
		exit()

	game_board()

def player2():
	game_board()

# Determines if input from user is an integer and is 1-9. Also if slot on board is taken by an "X" or an "O".
	try:
		global action1
		action1 = int(input("\nWhat is your move player 2? (1-9) "))
	except ValueError:
		action1 = None
		pass
	if action1 is not None:
		if int(action1) >= 0 and int(action1) <= 10:
			if type(game_move[int(action1) - 1]) == int:
				game_move[int(action1) - 1] = "O"
			else:
				game_board()
				player2()
		else:
			game_board()
			player2()
	else:
		game_board()
		player2()

# Win condition for player 2.
	if game_move[0] == "O" and game_move[1] == "O" and game_move[2] == "O" or game_move[3] == "O" and game_move[4] == "O" and game_move[5] == "O" or game_move[6] == "O" and game_move[7] == "O" and game_move[8] == "O" or game_move[0] == "O" and game_move[3] == "O" and game_move[6] == "O" or game_move[1] == "O" and game_move[4] == "O" and game_move[7] == "O" or game_move[2] == "O" and game_move[5] == "O" and game_move[8] == "O" or game_move[0] == "O" and game_move[4] == "O" and game_move[8] == "O" or game_move[2] == "O" and game_move[4] == "O" and game_move[6] == "O":
		game_board()
		print("Player 2 wins!")
		exit()



# Game start.
while type(game_move[0]) == int or type(game_move[1]) == int or type(game_move[2]) == int or type(game_move[3]) == int or type(game_move[4]) == int or type(game_move[5]) == int or type(game_move[6]) == int or type(game_move[7]) == int or type(game_move[8]) == int:
  player1()
  if type(game_move[0]) == int or type(game_move[1]) == int or type(game_move[2]) == int or type(game_move[3]) == int or type(game_move[4]) == int or type(game_move[5]) == int or type(game_move[6]) == int or type(game_move[7]) == int or type(game_move[8]) == int:
    player2()
print("\nNobody wins!")